main.go

```diff
@@ -32,54 +37,84 @@ func getConfig() config {
 	url := os.Getenv("JENKINS_BASE_URL")
 	githubSecret := os.Getenv("GITHUB_SECRET")
 	githubToken := os.Getenv("GITHUB_TOKEN")
+	integrationDirectory := os.Getenv("INTEGRATION_DIRECTORY")
+
+	// repos that require to be tested with specific integration branches
+	integrationBranchDependant :=
+		[]string{
+			"deployments",
+			"deviceadm",
+			"deviceauth",
+			"useradm",
+			"inventory",
+			"mender-api-gateway-docker",
+			"mender",
+		}
+
+	// if no env. variable is set, this is the default repo watch list
+	defaultWatchRepositories :=
+		[]string{
+			"deployments",
+			"deviceadm",
+			"deviceauth",
+			"inventory",
+			"useradm",
+			"integration",
+			"mender",
+			"mender-qa",
+			"mender-artifact",
+			"meta-mender",
+			"mender-api-gateway-docker"}
 
-	defaultWatchRepositories := "deployments,deviceadm,deviceauth,inventory,useradm,integration,mender,meta-mender"
 	watchRepositories := os.Getenv("WATCH_REPOS")
 
 	if len(watchRepositories) == 0 {
-		repositoryWatchList = strings.Split(defaultWatchRepositories, ",")
+		repositoryWatchList = defaultWatchRepositories
 	} else {
 		repositoryWatchList = strings.Split(watchRepositories, ",")
 	}
 
-	if username == "" {
+	switch {
+	case username == "":
 		panic("set JENKINS_USERNAME")
```

Comment:
avoid using panic, log.Fatal or even better - return those are errors.New and log.Fatal from main